zerver/lib/streams.py

```diff
@@ -781,8 +819,11 @@ def can_administer_channel(channel: Stream, user_profile: UserProfile) -> bool:
 
 
 def filter_stream_authorization(
-    user_profile: UserProfile, streams: Collection[Stream]
-) -> tuple[list[Stream], list[Stream]]:
+    user_profile: UserProfile, streams: Collection[Stream], is_subscribing_other_users: bool = False
+) -> tuple[list[Stream], list[Stream], list[Stream]]:
+    if len(streams) == 0:
+        return [], [], []
```

Comment:
The return type from this function is kinda terrible; we may want to rework it to return a `dataclass` as a follow-up.